client/api.go

```diff
@@ -195,6 +195,35 @@ func (c *Client) apiCreate(ctx context.Context, url string, o interface{}) (objJ
 	return objJSON, nil
 }
 
+func (c *Client) apiUpdate(ctx context.Context, url string, o interface{}) (objJSON []byte, err error) {
```

Comment:
You might consider reducing code by having apiCreate and apiUpdate call a new function that has extra parameter for HTTP method and an HTTP status code to check for - otherwise they look the same to me.